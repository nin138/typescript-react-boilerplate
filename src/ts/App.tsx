import * as React from 'react';
import { useState } from 'react';
import * as styles from '../css/app.css';

export const App = () => {
  const [text, setText] = useState('');
  return (
    <div className={styles.app}>
      <img height={'16px'} width={'16px'} src={'./images/logo.svg'} alt={'logo'} />
      {text || 'hello, world!'}
      <br />
      <input value={text} onChange={e => setText(e.target.value)} />
    </div>
  );
};
