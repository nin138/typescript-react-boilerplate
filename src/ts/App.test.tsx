import * as React from "react";
import { App } from "./App";
import renderer from "react-test-renderer";

test("App component", () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON()!;
  expect(tree).toMatchSnapshot();

  tree.children!.find(it => it.type === "input")!.props.onChange({ target: { value: "test text" } });
  // re-rendering
  tree = component.toJSON()!;
  expect(tree).toMatchSnapshot();
});
