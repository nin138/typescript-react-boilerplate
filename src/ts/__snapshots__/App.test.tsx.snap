// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component 1`] = `
<div>
  hello, world!
  <br />
  <input
    onChange={[Function]}
    value=""
  />
</div>
`;

exports[`App component 2`] = `
<div>
  test text
  <br />
  <input
    onChange={[Function]}
    value="test text"
  />
</div>
`;
